### DISCUSSION TITLE
Adding custom observer to send custom metrics to the logs

### DISCUSSION THREAD

**Discussion by Nikschavan** [Pierre-Lannoy/wp-decalog#66] [General]
I am using DataDog Events with Decalog, With the default observers, I want to add some more information to the event sent from decalog. 

For eg 

1. more information about the current environment apart from just the WP
2. Maybe change some of the default parameters sent by the `WordpressProcessor`

Is there a way to add a new log processor similar to `WordpressProcessor` and others and handle custom data in that processor? or any other ways to add more data about the current environment?

**Comment by Pierre-Lannoy** [comment]
Hi!

**Reply by Nikschavan** [reply]
👋🏼

**Comment by Nikschavan** [comment]
I was able to add this using the "Extended fields" that is available on the DataDog logger.
Since I am using `DECALOG_DEFAULT_LOGGERS` constant to set the logger from code, I am also able to set the extra fields dynamically.
